doc/source/cluster/kubernetes/examples/distributed-checkpointing-with-gcsfuse.md

```diff
@@ -0,0 +1,264 @@
+
+## Create a Kubernetes cluster on GKE
+
+Create a GKE cluster with the [GCSFuse CSI driver](https://cloud.google.com/kubernetes-engine/docs/how-to/persistent-volumes/cloud-storage-fuse-csi-driver) and [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) enabled, as well as a GPU node pool with 4 L4 GPUs:
+```
+export PROJECT_ID=<your project id>
+gcloud container clusters create kuberay-with-gcsfuse \
+    --addons GcsFuseCsiDriver \
+    --cluster-version=1.29.3-gke.1093000 \
+    --location=us-east4-c \
+    --machine-type=g2-standard-8 \
+    --num-nodes=4 \
+    --accelerator type=nvidia-l4,count=1,gpu-driver-version=latest \
+    --workload-pool=${PROJECT_ID}.svc.id.goog
+```
+
+## Install the KubeRay operator
+
+Follow [Deploy a KubeRay operator](kuberay-operator-deploy) to install the latest stable KubeRay operator from the Helm repository.
+The KubeRay operator Pod must be on the CPU node if you set up the taint for the GPU node pool correctly.
+
+## Configuring the GCS Bucket
+
+Create a GCS bucket that will be used as the remote filesystem.
+```
+gcloud storage buckets create gs://<your bucket name>
+```
+
+Create a Kubernetes ServiceAccount that will be used to grant the RayCluster access to mount the GCS bucket:
+```
+kubectl create serviceaccount pytorch-distributed-training
+```
+
+Bind the `roles/storage.objectUser` role to the Kubernetes service account and bucket IAM policy:
+```
+PROJECT_ID=<your project ID>
+PROJECT_NUMBER=<your project number>
+BUCKET=<your GCS BUCKET>
+gcloud storage buckets add-iam-policy-binding gs://${BUCKET} --member "principal://iam.googleapis.com/projects/${PROJECT_NUMBER}/locations/global/workloadIdentityPools/${PROJECT_ID}.svc.id.goog/subject/ns/default/sa/pytorch-distributed-training"  --role "roles/storage.objectUser"
+```
+
+See [Access Cloud Storage buckets with the Cloud Storage FUSE CSI driver](https://cloud.google.com/kubernetes-engine/docs/how-to/persistent-volumes/cloud-storage-fuse-csi-driver) for more details.
+
+## Deploy the RayJob
+
+Download the RayJob that executes all the steps documented in [Finetuning a Pytorch Image Classifier with Ray Train](https://docs.ray.io/en/latest/train/examples/pytorch/pytorch_resnet_finetune.html). The [source code](https://github.com/ray-project/kuberay/tree/master/ray-operator/config/samples/pytorch-resnet-image-classifier) is also in the KubeRay repository.
+
+```
+curl -LO https://raw.githubusercontent.com/ray-project/kuberay/master/ray-operator/config/samples/pytorch-resnet-image-classifier/ray-job.pytorch-image-classifier.yaml
+```
+
+Modify the RayJob by replacing all instances of the `GCS_BUCKET` placeholder with the Google Cloud Storage bucket you created earlier. Alternatively you can use `sed`:
+```
+sed -i "s/GCS_BUCKET/<your-bucket>/g" ray-job.pytorch-image-classifier.yaml
+```
+
+Deploy the RayJob
+```
+kubectl create -f ray-job.pytorch-image-classifier.yaml
```

Comment:
The worker's init container can't complete.

```sh
> kubectl describe pod lassifier-h4h2b-raycluster-ldzg2-worker-gpu-group-q6zxf

Events:
  Type     Reason     Age               From               Message
  ----     ------     ----              ----               -------
  Normal   Scheduled  8m4s              default-scheduler  Successfully assigned default/lassifier-h4h2b-raycluster-ldzg2-worker-gpu-group-q6zxf to gke-kuberay-with-gcsfuse-default-pool-845e7d9f-mhhv
  Normal   Pulling    8m3s              kubelet            Pulling image "rayproject/ray-ml:2.9.0-gpu"
  Normal   Pulled     3m27s             kubelet            Successfully pulled image "rayproject/ray-ml:2.9.0-gpu" in 4m36.355s (4m36.355s including waiting)
  Warning  Failed     87s               kubelet            Error: context deadline exceeded
  Normal   Pulled     2s (x8 over 87s)  kubelet            Container image "rayproject/ray-ml:2.9.0-gpu" already present on machine
  Warning  Failed     2s (x8 over 87s)  kubelet            Error: failed to reserve container name "wait-gcs-ready_lassifier-h4h2b-raycluster-ldzg2-worker-gpu-group-q6zxf_default_fb1d16f1-8425-4a73-8b83-2bbca5eebc19_0": name "wait-gcs-ready_lassifier-h4h2b-raycluster-ldzg2-worker-gpu-group-q6zxf_default_fb1d16f1-8425-4a73-8b83-2bbca5eebc19_0" is reserved for "a586844444d34eedee90c6efa2cdac54411e0eb711a5c721993a252697a88b15"

> kubectl logs lassifier-h4h2b-raycluster-ldzg2-worker-gpu-group-q6zxf -c wait-gcs-ready
Error from server (BadRequest): container "wait-gcs-ready" in pod "lassifier-h4h2b-raycluster-ldzg2-worker-gpu-group-q6zxf" is waiting to start: CreateContainerError
```

Reply:
The init container bug is related to https://github.com/ray-project/kuberay/issues/1130, this is why the example references a newer v1.29.3 GKE version . If you created with 1.29.1, it won't have the fix mentioned in the issue

Reply:
@kevin85421 I updated the gcloud command above to use `--release-channel=rapid` so you should be able to specify GKE version 1.29.3-gke.1093000 which fixes this issue. Can you try again with the updated version?